src/user/services/user.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { User } from '../entities/user.entity';
+import { Repository } from 'typeorm';
+
+@Injectable()
+export class UserService {
+  constructor(
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+  ) {}
+
+  async findOneByAttr(attr: string, value: any): Promise<User | null> {
+    return await this.userRepository.findOne({ where: { [attr]: value } });
```

Comment:
FYI: you can use findOneBy
Refer to https://g.co/gemini/share/5ff9a7be9445

example: 
// Simple match by ID
const user = await userRepository.findOneBy({ id: 123 });

Reply:
+

Reply:
Thanks guys! I changed them all. please check them again.